<?xml version="1.0" encoding="UTF-8" ?>
<launch>
    <arg name="segment_file" default="road_segments.pyc"/>
    <arg name="road_info_file" default="road_info_twoway.pyc"/>

    <node name="mag_shift" pkg="gps_to_path" type="get_mag_shift"
          respawn="true" respawn_delay="1.0"/>

    <node name="magnetometer_compass" pkg="compass" type="magnetometer_compass"
          respawn="true" respawn_delay="1.0" output="screen">
        <param name="publish_true_azimuth_enu_rad" type="bool" value="true"/>
        <param name="magnetic_model" type="str" value="wmm2020"/>
    </node>

    <node name="start_srv" pkg="road_crossing" type="start_srv"
          respawn="true" respawn_delay="1.0" output="screen"/>

    <node name="get_finish" pkg="road_crossing" type="get_finish_srv"
          respawn="true" respawn_delay="1.0" output="screen"/>

    <param name="road_crossing/road_file_name" type="str" value="$(dirname)/../data/$(arg segment_file)"/>

    <node name="get_segment" pkg="road_crossing" type="get_segment_srv"
          respawn="true" respawn_delay="1.0" output="screen">
    </node>
    
    <node name="get_suitability" pkg="road_crossing" type="get_suitability_srv"
          respawn="true" respawn_delay="1.0" output="screen"/>

    <param name="road_crossing/road_info_file" type="str" value="$(dirname)/../data/$(arg road_info_file)"/>

    <node name="get_road_info" pkg="road_crossing" type="get_road_info_srv"
          respawn="true" respawn_delay="1.0" output="screen">
    </node>
</launch>
